### ISSUE TITLE
microsoft.ad.group fails if group already exists

### ISSUE THREAD

**Issue by zimbeo** [ansible-collections/microsoft.ad#52]
##### SUMMARY
In the `microsoft.ad.group` module if I execute a playbook to add a computer to an existing security group it fails with the following error - `New-ADGroup failed: The specified group already exists`. I'm not sure if I'm just missing something in this new module compared to the old `community.windows.win_domain_group_membership` one but I am unable to solve this and can't find where I'm going wrong. My role is structured the exact same as one of the examples within the docs so I feel like this should "just work".

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`microsoft.ad.group`

##### ANSIBLE VERSION

```
ansible-playbook [core 2.14.2]
  config file = /runner/project/ansible.cfg
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /runner/requirements_collections:/home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible-playbook
  python version = 3.9.13 (main, Nov  9 2022, 13:16:24) [GCC 8.5.0 20210514 (Red Hat 8.5.0-15)] (/usr/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```pastebelow
Collection   Version
------------ -------
microsoft.ad 1.2.0
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = /etc/ansible/ansible.cfg
```

##### OS / ENVIRONMENT

- Ansible Automation Platform Controller 4.2.0 running on Red Hat Enterprise Linux release 8.8
- delegate_to host is Windows Server 2019, Version 1809

##### STEPS TO REPRODUCE

Here is the role `managed_servers_group`
```yaml
- name: Add server to group
  microsoft.ad.group:
    name: 'managed_windows_servers'
    scope: global
    members:
     add:
       - 'servername01$'
    state: present
  delegate_to: jumpbox
```

The playbook
```
- name: Provisioning Windows
  hosts: all
  gather_facts: no

  roles:
     - { role: managed_servers_group}
```

Executing
```
ansible-playbook -i myinventory main.yml -l servername01
```

##### EXPECTED RESULTS

The AD group doesn't currently have servername01 in it. I would expect it to be added in addition to the current members being maintained.

##### ACTUAL RESULTS

```paste below
The full traceback is:
The specified group already exists
At line:963 char:29
+                 $adObject = & $newCommand @newParams @adParams
+                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (CN=managed_wind...DC=domain,DC=com:String) [New-ADGroup], ADException
    + FullyQualifiedErrorId : ActiveDirectoryServer:1318,Microsoft.ActiveDirectory.Management.Commands.NewADGroup

ScriptStackTrace:
at Invoke-AnsibleADObject, <No file>: line 963
at <ScriptBlock>, <No file>: line 211
fatal: [servername01 -> jumpbox]: FAILED! => {
    "changed": false,
    "distinguished_name": null,
    "invocation": {
        "module_args": {
            "attributes": {
                "add": {},
                "remove": {},
```

**Comment by jborean93** [comment]
I believe this is a duplicate of https://github.com/ansible-collections/microsoft.ad/issues/44#issuecomment-1610665481. Essentially the old `win_domain_*` modules had a mixture of lookup behaviours whereas the new modules in this collection treat `name` as the LDAP `cn` entry in the path specified. If the group `managed_windows_servers` already exists but not in the default path used to create groups, it will treat the object to manage as not being present and will attempt to create it again.

TLDR: you need to specify `path` to state where the group should exist in or else it won't be able to find it. Unfortunately as my link states you should also be able to use `identity` but I need to update the logic to better handle `path` not being set with `identity`.
